server/sonar-server/src/main/java/org/sonar/server/plugins/ws/UninstallPluginsWsAction.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.plugins.ws;
+
+import com.google.common.base.Predicate;
+import org.sonar.api.platform.PluginMetadata;
+import org.sonar.api.platform.PluginRepository;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.server.plugins.ServerPluginJarsInstaller;
+
+import javax.annotation.Nullable;
+
+import static com.google.common.collect.Iterables.find;
+import static java.lang.String.format;
+
+/**
+ * Implementation of the {@code uninstall} action for the Plugins WebService.
+ */
+public class UninstallPluginsWsAction implements PluginsWsAction {
+  private static final String PARAM_KEY = "key";
+
+  private final PluginRepository pluginRepository;
+  private final ServerPluginJarsInstaller pluginJarsInstaller;
+
+  public UninstallPluginsWsAction(PluginRepository pluginRepository, ServerPluginJarsInstaller pluginJarsInstaller) {
+    this.pluginRepository = pluginRepository;
+    this.pluginJarsInstaller = pluginJarsInstaller;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    WebService.NewAction action = controller.createAction("uninstall")
+      .setPost(true)
+      .setDescription("Uninstalls the plugin specified by its key.")
+      .setHandler(this);
+
+    action.createParam(PARAM_KEY)
+      .setDescription("The key identifying the plugin to uninstall")
+      .setRequired(true);
```

Comment:
initially thought about providing an example value, but for a plugin key, I didn't see the value